DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_config.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import asyncio
+
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.lib.ethtool.ethtool import Ethtool
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_dev_groups_from_config,
+    tgen_utils_traffic_generator_connect,
+    tgen_utils_update_l1_config
+)
+
+pytestmark = [pytest.mark.suite_functional_l1,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures("cleanup_bridges", "cleanup_tgen")]
+
+
+@pytest.mark.parametrize("settings", ["autodetect", "autoneg"])
+async def test_l1_settings_(testbed, settings):
+    """
+    Test Name: L1 settings verification
+    Test Suite: suite_functional_l1
+    Test Overview: Verify L1 settings on dut port
+    Test Procedure:
+    1. Create bridge entity
+    2. Enslave ports to the created bridge entity
+    3. Set up port duplex, speed
+    4. Verify port mode and speed
+    5. Verify "Link Partner Advertised Auto-negotiation" is set to `Yes`
+    6. Verify "Advertised Auto-negotiation" is set to `Yes`
+    7. Verify "Link Partner Advertised Link Modes" is set to configured speed/duplex
+    8. Verify "Advertised Link Modes" is set to configured speed/duplex
+    """
+
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 1)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgen connections")
+        return
+    device_host_name = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0][0]
+    port = tgen_dev.links_dict[device_host_name][1][0]
+    speed = 10
+    duplex = "full"
+    advertise = "0x002"
+
+    options = {"autodetect":
+               {
+                   "devname": port,
+                   "speed": speed,
+                   "duplex": "full"},
+               "autoneg":
+               {
+                   "devname": port,
+                   "autoneg": "on",
+                   "advertise": advertise}
+               }
```

Comment:
this is hard to read
```suggestion
    options = {
        "autodetect": {
            "devname": port,
            "speed": speed,
            "duplex": "full",
        },
        "autoneg": {
            "devname": port,
            "autoneg": "on",
            "advertise": advertise,
        },
    }
```